Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectTeams.cs

```diff
@@ -594,7 +594,7 @@ private static bool SetGroupSecurity(PnPMonitoredScope scope, Team team, string
         /// <returns>Whether the Channels have been provisioned or not</returns>
         private static bool SetTeamChannels(PnPMonitoredScope scope, TokenParser parser, Team team, string teamId, string accessToken)
         {
-            if (team.Channels != null)
+            if (team.Channels.Count > 0)
```

Comment:
Hi, what is the purpose of this test? If the Channels collection is not null but still empty the foreach clause will be skipped. On the contrary if Channels is null and we test the Count property, we will get a null reference exception.